src/validation.cpp

```diff
@@ -3114,13 +3112,12 @@ void CChainState::ProcessLoanEvents(const CBlockIndex* pindex, CCustomCSView& ca
         if( (pindex->nHeight - loanToken.creationHeight) % priceHeight == 0)
         {
             auto updateLoanToken = loanToken;
-            auto aggPrice = GetAggregatePrice(cache, updateLoanToken.priceFeedId.first, updateLoanToken.priceFeedId.second, pindex->nHeight);
-            if(!aggPrice){
-                return true;
-            }
-            updateLoanToken.activePrice = loanToken.nextPrice;
-            updateLoanToken.nextPrice = aggPrice;
-            cache.LoanUpdateLoanToken(updateLoanToken, key);
+
+            auto priceFeed = cache.GetPriceFeedData(loanToken.priceFeedId);
+            assert(priceFeed);
+            priceFeed.val->activePrice = priceFeed.val->nextPrice;
+            priceFeed.val->nextPrice = GetAggregatePrice(cache, loanToken.priceFeedId.first, loanToken.priceFeedId.second, pindex->nTime);
+            cache.UpdatePriceFeed(loanToken.priceFeedId, priceFeed);
```

Comment:
This is setting the priceFeed based on the loan token creation height? What if there are multiple loan tokens using the same price feed? Won't it get set multiple times?